extensions-contrib/materialized-view-selection/src/main/java/org/apache/druid/query/materializedview/DataSourceOptimizerMonitor.java

```diff
@@ -50,15 +52,15 @@ public boolean doMonitor(ServiceEmitter emitter)
       emitter.emit(builder.build("/materialized/view/query/hitRate", stat.getHitRate()));
       emitter.emit(builder.build("/materialized/view/select/avgCostMS", stat.getOptimizerCost()));
       Map<String, Long> derivativesStats = stat.getDerivativesHitCount();
-      for (String derivative : derivativesStats.keySet()) {
-        builder.setDimension("derivative", derivative);
-        emitter.emit(builder.build("/materialized/view/derivative/numSelected", derivativesStats.get(derivative)));
+      for (Map.Entry<String, Long> entry : derivativesStats.entrySet()) {
+        builder.setDimension("derivative", entry.getKey());
+        emitter.emit(builder.build("/materialized/view/derivative/numSelected", derivativesStats.get(entry.getKey())));
       }
       final ServiceMetricEvent.Builder builder2 = new ServiceMetricEvent.Builder();
       builder2.setDimension("dataSource", stat.getBase());
-      for (Set<String> fields : stat.getMissFields().keySet()) {
-        builder2.setDimension("fields", fields.toString());
-        emitter.emit(builder2.build("/materialized/view/missNum", stat.getMissFields().get(fields).get()));
+      for (Map.Entry<Set<String>, AtomicLong> fields : stat.getMissFields().entrySet()) {
+        builder2.setDimension("fields", fields.getKey().toString());
+        emitter.emit(builder2.build("/materialized/view/missNum", stat.getMissFields().get(fields.getKey()).get()));
```

Comment:
Should be `e.getValue()` instead of `stat.getMissFields().get(fields.getKey())`.

@hpandeycodeit, sorry, I cannot spend so much time fixing every line in your code again and again.